src/builder.rs

```diff
@@ -1297,11 +1570,16 @@ mod tests {
             pod_spec
                 .init_containers
                 .as_ref()
-                .unwrap()
+                .and_then(|containers| containers.get(0).as_ref().map(|c| c.name.clone())),
+            Some("init_containername".to_string())
+        );
+
+        assert_eq!(
+            pod_spec.volumes.as_ref().and_then(|volumes| volumes
                 .get(0)
-                .unwrap()
-                .name,
-            "init_containername"
+                .as_ref()
+                .and_then(|volume| volume.config_map.as_ref().and_then(|cm| cm.name.clone()))),
```

Comment:
Now that you're inside of the `Option::and_then` you can use `?` to short-circuit away `None` rather than nesting `Option::and_then`s.